lib/active_merchant/billing/gateways/agms.rb

```diff
@@ -0,0 +1,322 @@
+# Agms Gateway uses nokogiri to build the xml request
+require 'nokogiri'
+
+module ActiveMerchant #:nodoc:
+  module Billing #:nodoc:
+    class AgmsGateway < Gateway
+      self.test_url = 'https://gateway.agms.com/roxapi/agms.asmx'
+      self.live_url = 'https://gateway.agms.com/roxapi/agms.asmx'
+
+      self.supported_countries = ['US']
+      self.default_currency = 'USD'
+      self.supported_cardtypes = [:visa, :master, :american_express, :discover, :dinner_club, :jcb]
+
+      self.homepage_url = 'http://onlinepaymentprocessing.com/'
+      self.display_name = 'AGMS Gateway'
+
+      STANDARD_ERROR_CODE_MAPPING = {
+        "2" => STANDARD_ERROR_CODE[:card_declined],
+        "10" => STANDARD_ERROR_CODE[:processing_error],
+        "20" => STANDARD_ERROR_CODE[:processing_error],
+        "30" => STANDARD_ERROR_CODE[:processing_error],
+      }
+
+      def initialize(options={})
+        requires!(options, :login, :password)
+        # Assign options to Class Instance
+        @options = options
+        super
+      end
+
+      def purchase(money, payment, options={})
+        # Purchase is sale tranaction, we create an array of params and then call the processing function
```

Comment:
High-level per-method comments are fine, but please strip out the intra-method comments.